ShellPkg/Application/Shell/ShellParametersProtocol.c

```diff
@@ -403,29 +403,26 @@ CreatePopulateInstallShellParametersProtocol (
   //
   // Populate the 3 faked file systems...
   //
+  (*NewShellParameters)->StdIn  = &FileInterfaceStdIn;
+  (*NewShellParameters)->StdOut = &FileInterfaceStdOut;
+  (*NewShellParameters)->StdErr = &FileInterfaceStdErr;
   if (*RootShellInstance) {
-    (*NewShellParameters)->StdIn  = &FileInterfaceStdIn;
-    (*NewShellParameters)->StdOut = &FileInterfaceStdOut;
-    (*NewShellParameters)->StdErr = &FileInterfaceStdErr;
-    Status                        = gBS->InstallProtocolInterface (
-                                           &gImageHandle,
-                                           &gEfiShellParametersProtocolGuid,
-                                           EFI_NATIVE_INTERFACE,
-                                           (VOID *)(*NewShellParameters)
-                                           );

```

Comment:
Please explain how this change is related to the symptom you are addressing, and why this change in behavior (not copying the existing ones) is guaranteed to be correct in all cases.